scripts/layouts_tool/layout.h

```diff
@@ -0,0 +1,32 @@
+#ifndef LAYOUT_H
+#define LAYOUT_H
+
+#include <QString>
+#include <QX11Info>
+#include <QStringList>
+#include "defs.h"
+
+struct KbdKeyChar {
+    char16_t character;
+    bool is_dead;
+};
+
+typedef const KbdKeyChar (*KeyboardLayoutPointer)[2];
```

Comment:
Layouts in `layouts.cpp` are read-only [`static const KbdKeyChar`](https://github.com/mixxxdj/mixxx/pull/966/files#diff-cb825e7999197c1e920731037499e800R27), which I think is a good idea so that nobody can modify the `KbdKeyChar` objects by accident.

In order to return these pointers by [`getLayout()`](https://github.com/mixxxdj/mixxx/pull/966/files#diff-cb825e7999197c1e920731037499e800R972) we need to have the `const` keyword there.

```c++
// This wouldn't work
const KeyboardLayoutPointer getLayout(std::string layoutName);
```